### ISSUE TITLE
Setting an AAAA record for a subdomain sets A record to local

### ISSUE THREAD

**Issue by charlydelta** [mail-in-a-box/mailinabox#1369]
My situation is the following: I have a domain, lets say example.com, and two servers, `aserver` and `bserver`.

- `aserver` is reachable through `aserver.example.com`
- `bserver` is reachable through `bserver.example.com`
- `bserver` is a mailinabox. I exclusively use the Custom DNS functionality of this machine
- `aserver` has a dynamic IP address, so theres a Python script running that checks the IP and calls the API from mailinabox to update the DNS records of `aserver.example.com`

**Previously**
Everything worked as expected, since I had a IPv4 address and kept updating the A record. Worked like a charm. For each service I have, for example `cloud.example.com` is just made an CNAME record to `aserver.example.com` and it worked

**Now**
My ISP changed the connection to be IPv6 only. So I changed everything to AAAA records, but I have major issues with this setup and mailinabox

- I can set the AAAA record, but even though there is no A record listed in the Custom DNS page, it appears that an A record is implicitly set pointing to the mailinabox' IP address

=>

- I cannot remove the A record, having an AAAA record **only**
- Depending on how the client resolves the DNS, I either get a stable connection with AAAA, or it will prefer the A record (even though IPv6 is available) and it will prompt security warnings since the SSL certificates are those from  `bserver`, not those from `aserver`.

I hope that my problem is comprehensible. Ask me anything if there's something unclear.

**Comment by zoof** [comment]
I've experienced something similar although my ISP still assigns an IPv4 address, the curl command only returns the IPv6 address so I manually set the IPv4 address.  Is there any way to have MIAB fetch the IPv4 address when rtype is set to A?  Or should I only be using IPv6 at this point?

**Comment by charlydelta** [comment]
The problem described here is that I don't want an IPv4 address to be set at all, because I have none. However, MIAB's DNS implicitly sets one (to itself), so some of my devices prefer it over the AAAA record and get bad connections.

**Comment by jvolkenant** [comment]
Specifying with PUT request does not result in just an AAAA record being created?

curl -X PUT -d "2001::DE:AD:BE:EF" https://example.com/admin/dns/custom/custom.example.com/AAAA --user user:pass

**Comment by charlydelta** [comment]
> Specifying with PUT request does not result in just an AAAA record being created?

Exactly, for me it also creates an A record pointing to the MiaB's IPv4. However, this record is not shown in the admin panel, I just realised it when I used DNS checker tools like mxtoolbox.

**Comment by jvolkenant** [comment]
Setting  an AAAA record with the curl does not add an ipv4 address:

root@m:~# dig +short aaaatest.example.com A
root@m:~# dig +short aaaatest.example.com AAAA
2001::de:ad:be:ef

Even if I add a wildcard dns entry with * it does not resolve an A record if an AAAA exists (but no A record).

Not really sure

**Comment by charlydelta** [comment]
I really don't know why it behaves for me like that. If i dig the A record I get the IP of the MiaB. I can even force a delete with 'curl -X DELETE` and the A record will still remain. It is undeletable. What can I do to provide better information to debug this issue?

**Comment by charlydelta** [comment]
May I ask what DNS server mailinabox uses or what command I could run on the machine to manually remove that A record?

**Comment by yodax** [comment]
Miab uses nsd

**Comment by charlydelta** [comment]
Thank you @yodax. I just took a look at the config files in `/etc/nsd/zones` and indeed, there is an A record explicitely set which is not visisble in the Miab UI. I will now try to find out how to cleanly remove that.

**Comment by charlydelta** [comment]
No matter what, Miab will always overwrite the changes. I tried to remove the A record from the conf file, of no avail. I tried to make the change, run `nsd-control reload <myzone> && nsd-control notify <myzone>`, but the changes didn't apply. After that, I made a change in the Miab control panel but the A record was then re-added into the zone file. 

The A record is, however, not listed in `/home/user-data/dns/custom.yaml`. I then thought that maybe I do not update the DNS cache properly, so I tried to see how the Miab control source code looks like and stumbled across [this line](https://github.com/mail-in-a-box/mailinabox/blob/master/management/dns_update.py#L229).

It seems like Miab sets the A record, or is this unrelated? If not, how come that the A record is not set for @jvolkenant?

**Comment by JoshData** [comment]
See also https://discourse.mailinabox.email/t/forum-not-reachable-over-ipv6/3234.
